libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -503,8 +504,15 @@ export class TurnContext {
                 // Append activities to buffer
                 const responses: ResourceResponse[] = [];
                 output.forEach((a) => {
-                        this.bufferedReplyActivities.push(a);
-                        responses.push({ id: undefined });
+                    this.bufferedReplyActivities.push(a);
+
+                    // Ensure the TurnState has the InvokeResponseKey, since this activity
+                    // is not being sent through the adapter, where it would be added to TurnState.
+                    if (a.type === ActivityTypes.InvokeResponse) {
```

Comment:
This is, I think, equivalent to https://github.com/microsoft/botbuilder-dotnet/pull/4845/files#diff-391e4084826390583f2af2b2d27d0d0add457734d88f05e817e826af88a2b41aR328

Reply:
The JS code looks good 👍 

@gabog, @johnataylor do we need to do a TryAdd in .NET or some sort of TrySet?

Reply:
We want to do a last-write wins in .NET if I'm correctly recall the implementation around TeamsActivityHandler. `.Add()` (depending on the class/implementation) normally throws an exception iirc.